src/libguac/client.c

```diff
@@ -701,6 +739,116 @@ int guac_client_owner_supports_required(guac_client* client) {
     
 }
 
+/**
+ * A callback function that is invokved by guac_client_owner_notify_join() to
+ * notify the owner of a connection that another user has joined the
+ * connection, returning zero if the message is sent successfully, or non-zero
+ * if an error occurs.
+ *
+ * @param user
+ *     The user to send the notification to, which will be the owner of the
+ *     connection.
+ *
+ * @param data
+ *     The data provided to the callback, which is the user that is joining the
+ *     connection.
+ *
+ * @return
+ *     Zero if the message is sent successfully to the owner, otherwise
+ *     non-zero, cast as a void*.
+ */
+static void* guac_client_owner_notify_join_callback(guac_user* user, void* data) {
+
+    const guac_user* joiner = (const guac_user *) data;
+
+    if (user == NULL)
+        return (void*) ((intptr_t) -1);
+
+    char* owner = "owner";
+    if (user->info.name != NULL)
+        owner = (char *) user->info.name;
+
+    char* joinName = "anonymous";
+    if (joiner->info.name != NULL)
+        joinName = (char *) joiner->info.name;
+
+    guac_user_log(user, GUAC_LOG_DEBUG, "Notifying owner \"%s\" of \"%s\" joining.", owner, joinName);
+    
+    /* Send required parameters to owner. */
+    const char* args[] = { (const char*)joinName, NULL };
```

Comment:
For `GUAC_MESSAGE_USER_JOINED` and `GUAC_MESSAGES_USER_LEFT`, thoughts on including the unique ID automatically assigned by guacd? For example:

```
const char* args[] = { joiner->user_id, (const char*) joinName, NULL };
```

In the event that an implementation assigns the same human-readable name to two or more users of a connection, including the unique ID would allow the receiving client to reliably differentiate between them.

Reply:
Yeah, I like that idea - makes sense to me.

Reply:
One follow-up question, here, @mike-jumper - are you thinking that the UUID would be displayed as part of the message the connection owner, or just logged somewhere? In the past we've avoided exposing some of the internals like UUIDs to the end user - not sure if this is a good place to make an exception?

Reply:
Nope - definitely should keep internals internal. My thinking was that client implementations may need to reliably differentiate between distinct users of a session, such as to provide a list of current users, regardless of whether their human-readable names match. Including the UUID would provide the necessary information for implementations to do that.

Reply:
Server-side is implemented - should send the `user_id` value on. Just have to update things on the client-side, now, as well.